src/main/java/ru/otus/hw1/dao/LocalFileTestDataService.java

```diff
@@ -0,0 +1,104 @@
+package ru.otus.hw1.dao;
+
+import lombok.extern.java.Log;
+import org.apache.commons.io.FilenameUtils;
+import ru.otus.hw1.domain.Choice;
+import ru.otus.hw1.domain.Question;
+import ru.otus.hw1.domain.Test;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Log
+public class LocalFileTestDataService implements TestDataService {
+
+    private final Path path;
+    private final String globTemplate;
+    private final String separator;
+    private final String correctSuffix;
+    private volatile Map<String, Path> tests;
+
+    public LocalFileTestDataService(Path path,
+                                    String globTemplate,
+                                    String separator,
+                                    String correctSuffix) {
+        this.path = path;
+        this.globTemplate = globTemplate;
+        this.separator = separator;
+        this.correctSuffix = correctSuffix;
+        tests = new HashMap<>();
+        loadTestList();
+    }
+
+    @Override
+    public Set<String> getAvailTests() {
+        if (tests.keySet()
+                 .size() > 0) {
+            return tests.keySet();
+        }
+        return null;
+    }
+
+    @Override
+    public Test getTest(String testName) {
+        if (testName == null
+            || "".equals(testName)
```

Comment:
Есть метод String.isEmpty :) ну или можно размер строки проверить.

Reply:
```suggestion
            || "".equals(testName)
```
Спасибо за подсказку, забыл совсем про этот метод. Наверно можно было бы даже использовать `isBlank()`, но наверно ничто не мешает завести файл с именем `     .csv`.
Заменил метода на `isEmpty()`